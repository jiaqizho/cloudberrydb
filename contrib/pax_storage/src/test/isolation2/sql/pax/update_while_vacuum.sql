-- @Description Ensures that an update before a vacuum operation is ok
-- 
DROP TABLE IF EXISTS pax_tbl;
CREATE TABLE pax_tbl (a INT, b INT);
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;

DELETE FROM pax_tbl WHERE a < 12;
1: BEGIN;
1: SELECT COUNT(*) FROM pax_tbl;
1>: UPDATE pax_tbl SET b=1 WHERE a > 0;COMMIT;
2: VACUUM pax_tbl;
1<:
1: SELECT COUNT(*) FROM pax_tbl;
3: INSERT INTO pax_tbl VALUES (0);
