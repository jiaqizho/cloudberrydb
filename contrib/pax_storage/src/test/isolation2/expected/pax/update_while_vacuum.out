-- @Description Ensures that an update before a vacuum operation is ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
CREATE TABLE pax_tbl (a INT, b INT);
CREATE
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;
INSERT 100

DELETE FROM pax_tbl WHERE a < 12;
DELETE 11
1: BEGIN;
BEGIN
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 89    
(1 row)
1>: UPDATE pax_tbl SET b=1 WHERE a > 0;COMMIT;  <waiting ...>
2: VACUUM pax_tbl;
VACUUM
1<:  <... completed>
UPDATE
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 89    
(1 row)
3: INSERT INTO pax_tbl VALUES (0);
INSERT 1
